File: src/main/java/com/dsm/pick/domains/domain/Notice.java
package com.dsm.pick.domains.domain;

import javax.persistence.*;
import java.sql.Timestamp;

@Entity
@Table(name = "notice")
public class Notice {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "notice_id")
    private int id;

    @Column(name = "content")
    private String content;

    @ManyToOne
    @JoinColumn(name = "admin_id", referencedColumnName = "id")
    private Admin admin;

    @Column(name = "category")
    private String category;

    @Column(name = "created_at")
    private Timestamp date;

    public Notice() {}
    public Notice(int id, String content, Admin admin, String category, Timestamp date) {
        this.id = id;
        this.content = content;
        this.admin = admin;
        this.category = category;
        this.date = date;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public Admin getAdmin() {
        return admin;
    }

    public void setAdmin(Admin admin) {
        this.admin = admin;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public Timestamp getDate() {
        return date;
    }

    public void setDate(Timestamp date) {
        this.date = date;
    }
}

File: src/main/kotlin/com/dsm/pick/domain/Notice.kt
package com.dsm.pick.domain

import com.dsm.pick.domain.attribute.Category
import com.dsm.pick.domain.converter.CategoryConverter
import java.sql.Timestamp
import javax.persistence.*

@Entity
@Table(name = "notice")
class Notice(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "notice_id")
    var id: Int? = null,

    @Column(name = "content")
    val content: String,

    @ManyToOne
    @JoinColumn(name = "admin_id", referencedColumnName = "id")
    val admin: Admin,

    @Column(name = "category")
    @Convert(converter = CategoryConverter::class)
    val category: Category,

    @Column(name = "created_at")
    val date: Timestamp,
) {

    override fun toString(): String {
        return "Notice(id=$id, content='$content', admin=$admin, category=$category, date=$date)"
    }
}